### PR TITLE
feat: Add Docker setup for easier developer environment configuration

### PR DESCRIPTION
**Pull request by ezhil56x**
## Description
- Added `docker-compose.yml` which can be used to start and stop the containers
- Environment variables which start with `NEXT` are required at build time so that should be added in `web/Dockerfile`
- `isaac-web` container uses the `web/.env.example`, these variables can be added in `.env.example`
- `isaac-api` container uses the `api/app/.env.example`, these variables can be added in `.env.example`
- Added support for Supabase CLI, with which local deployment of  Supabase is possible and added migrations. So that the database schema will be added

/claim #53

### CODE REVIEW

**Review by ologbonowiwi — CHANGES_REQUESTED** [review]
IMO we need also an update on supabase guide (following the example from capgo on the issue) and preferably a service on docket-compose.yml that runs supabase

**Comment by ezhil56x** [comment]
> IMO we need also an update on supabase guide (following the example from capgo on the issue) and preferably a service on docket-compose.yml that runs supabase

Will add a commit asap

**Comment by ologbonowiwi** [comment]
> Will add a commit asap

thank you! if possible (I don't know if it is), run also the necessary migrations, so the users will not have to run it by themselves

**Comment by eimenhmdt** [comment]
agree with @ologbonowiwi

**Comment by ezhil56x** [comment]
@ologbonowiwi @eimenhmdt Required changes have been done. Can you review once?

File: README.md [hunk 1]
```diff
@@ -34,7 +34,43 @@ We're on a mission to make Isaac the go-to tool for researchers, scientists, and
...
 
-## Installation Steps
+## Installation Steps (Docker)
+
+1. Install Supabase CLI from [here](https://supabase.com/docs/guides/cli/getting-started)
+
+2. Setup Supabase locally and migrate database schema by running the following command
+
+```bash
+supabase start
```
Comment (ologbonowiwi) [inline README.md:44]: I recommend to set examples using `npx supabase` so the users does not need to install it globally

Comment (ezhil56x) [reply README.md]: Done 👍🏻

File: README.md [hunk 2]
```diff
@@ -34,7 +34,43 @@ We're on a mission to make Isaac the go-to tool for researchers, scientists, and
 - Python 3.x
 - Supabase Account
 
-## Installation Steps
+## Installation Steps (Docker)
```
Comment (ologbonowiwi) [inline README.md:37]: as we're not using Docker for this, we can remove it
```suggestion
## Installation Steps
```

Comment (ologbonowiwi) [reply README.md:37]: ohh I see there are stuff related to the repo and docker also. maybe should we split then? do one part only to set up supabase

Comment (ologbonowiwi) [reply README.md:37]: actually, you can leave as it is

**Review by ologbonowiwi — APPROVED** [review]
LGTM

File: README.md
Comment (eimenhmdt) [inline README.md:1]: Change to: "Installation using Docker (Recommended)

File: README.md [hunk 3]
```diff
@@ -34,7 +34,43 @@ We're on a mission to make Isaac the go-to tool for researchers, scientists, and
...
+
+1. Install Supabase CLI from [here](https://supabase.com/docs/guides/cli/getting-started)
```
Comment (eimenhmdt) [inline README.md:39]: "Install the Supabase CLI following the instructions here"

File: README.md [hunk 4]
```diff
@@ -34,7 +34,43 @@ We're on a mission to make Isaac the go-to tool for researchers, scientists, and
...
+
+5. Fork the repo from [here](https://github.com/aietal/isaac/fork)
+
+6. Clone the forked repo to your local machine
+
+```bash
+git clone https://github.com/aietal/isaac.git
+```
+
+7. Add the other required environment variables to `api/app/.env.example`, `web/.env.example` and `web/Dockerfile`
```
Comment (eimenhmdt) [inline README.md:65]: Why to .env.example? The idea is to replace the .env.example files with actual .env files containing the environment vars.

File: README.md [hunk 5]
```diff
@@ -190,6 +226,10 @@ create table comments (
   updated_at timestamp default now()
 );
 
+CREATE TYPE upload_processing_status AS ENUM ('pending', 'processing', 'completed');
+
+create extension vector with schema extensions;
```
Comment (eimenhmdt) [inline README.md:231]: Change to: 

`create extension vector with schema public;`

File: api/app/.env.example
Comment (eimenhmdt) [inline api/app/.env.example:1]: move this to from api/app to api/

**Review by eimenhmdt — CHANGES_REQUESTED** [review]

File: api/app/.env.example
Comment (ezhil56x) [reply api/app/.env.example:1]: Done 👍🏻

File: README.md (hunk 4 above)
Comment (ezhil56x) [reply README.md]: `.env` is ignored by `.gitignore`. If required we can add another point to copy to `.env`

File: README.md (hunk 3 above)
Comment (ezhil56x) [reply README.md]: Done 👍🏻

File: README.md
Comment (ezhil56x) [reply README.md:1]: Done 👍🏻

File: README.md (hunk 5 above)
Comment (ezhil56x) [reply README.md:231]: This doesn't seem to work I have already tried, throws error

```
ERROR: type "extensions.vector" does not exist (SQLSTATE 42704)
At statement 12: create table document_embeddings (            
  id bigint not null primary key,                              
  content text,                                                
  metadata jsonb,                                              
  embedding extensions.vector(1536),                           
  upload_id uuid references uploads (id),                      
  searchable_content tsvector,                                 
  updated_at timestamp default now()                           
) 
```

**Comment by ezhil56x** [comment]
@eimenhmdt Any update or changes required?

**Review by eimenhmdt — APPROVED** [review]

**Comment by rishi-raj-jain** [comment]
```
create or replace function public.handle_new_user() 
returns trigger as $$
begin
  insert into public."profile" (id, email, has_seen_community_banner, editor_language)
  values (new.id, new.email, false, "English");
  return new;
end;
$$
language plpgsql
security definer
set search_path = public
;

create trigger on_auth_user_created
  after insert on auth.users
  for each row execute procedure public.handle_new_user();
```

fails to work

**Pull request merged**